CONTRIBUTING.md

```diff
@@ -0,0 +1,316 @@
+
+## Reporting Bugs
+
+Reporting bugs is a great way for anyone to help improve Swift.
+The open source Swift project uses GitHub Issues for tracking bugs.
+
+If a bug can be reproduced only within an Xcode project or a playground, or if the bug is associated with an Apple NDA, please file a report to Apple’s [bug reporter](https://bugreport.apple.com/) instead.
+When opening an issue, please include the following:
+
+- **A concise description of the problem**.
+If the issue is a crash, include a stack trace.
+Otherwise, describe the behavior you were expecting to see, along with the behavior you actually observed.
+
+- **A reproducible test case**.
+Double-check that your test case reproduces the issue.
+A relatively small sample (roughly within 50 lines of code) is best pasted directly into the description; a larger one may be uploaded as an attachment.
+Consider reducing the sample to the smallest amount of code possible; a smaller test case is easier to reason about and more appealing to сontributors.
+
+- **A description of the environment that reproduces the problem**.
+Include information about the Swift compiler’s version, the deployment target (if explicitly set) and your platform.
+
+Because Swift is under very active development, we receive a lot of bug reports.
+Before opening a new issue, take a moment to [browse our existing issues](https://github.com/swiftlang/swift/issues) to reduce the chance of reporting a duplicate.
+
+## Good First Issues
+
+Good first issues are bugs, ideas, and tasks that are intended to be accessible for contributors that are new to working on the Swift project, and even new to the patterns and concepts behind subprojects such as the Swift compiler.
+Good first issues are decorated with a corresponding label and are most easily found by visiting `github.com/apple/<repository>/contribute`.
+An example for the main Swift repository is [github.com/apple/swiftlang/contribute](https://github.com/swiftlang/swift/contribute).
+They are expected to be low-priority and of modest scope, and not require excessive refactoring, research, or debugging — rather, they should encourage newcomers to dip their toes in some part of Swift, learn more about it, and make a real contribution.
+
+Anyone with [commit access](https://www.swift.org/contributing/#commit-access) and insight into a particular area is welcome and encouraged to pin down or think up good first issues.
+
+## Contributing Code
+
+If you are interested in:
+
+- Contributing fixes and features to the compiler: See our
+  [How to Submit Your First Pull Request guide](/docs/HowToGuides/FirstPullRequest.md).
+- Building the compiler as a one-off: See our [Getting Started guide][].
+- Building a toolchain as a one-off: Follow the [Getting Started guide][]
+  up until the "Building the project" section. After that, follow the
+  instructions in the [Swift Toolchains](/README.md#swift-toolchains).
```

Comment:
I don’t think that a new contributor would want to build an entire toolchain. Doing so is its own can of worms and it’s rarely required. So I’d just remove the bullet. (You should be able to inline the Getting started guide link again as well.

Reply:
removed bullet and inlined the link to the getting started docs